### PR TITLE
[ownership] Move OME after CopyForwarding

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
This may break pass restart in the interim, since OME is a module pass and is being moved into a function pass pipeline.

File: lib/SILOptimizer/PassManager/PassPipeline.cpp [hunk 1]
```diff
@@ -482,10 +486,6 @@ static void addHighLevelFunctionPipeline(SILPassPipelinePlan &P) {
   // FIXME: update EagerSpecializer to be a function pass!
   P.addEagerSpecializer();
 
-  // We earlier eliminated ownership if we are not compiling the stdlib. Now
-  // handle the stdlib functions.
-  P.addNonTransparentFunctionOwnershipModelEliminator();
-
   addFunctionPasses(P, OptimizationLevelKind::HighLevel);
```
Comment (gottesmm) [inline lib/SILOptimizer/PassManager/PassPipeline.cpp:489]: Can you add a note here that we are lowering ownership in side addFunctionPasses. I am just trying to be very careful/explicit during the transition.

**Comment by meg-gupta** [comment]
@swift-ci benchmark

File: lib/SILOptimizer/PassManager/PassPipeline.cpp (hunk 1 above)
Comment (meg-gupta) [reply lib/SILOptimizer/PassManager/PassPipeline.cpp:490]: Done

**Comment by meg-gupta** [comment]
Converting the OME to a function pass is not an option. Doing so, we may end up inlining ossa functions to non-ossa functions and vice versa due to callgraph scc cycles. Added a comment to document this interim issue.

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test source compatibility

**Comment by meg-gupta** [comment]
@swift-ci smoke test OS X platform

**Comment by meg-gupta** [comment]
@swift-ci smoke test OS X platform

**Comment by meg-gupta** [comment]
@swift-ci test OS X platform

**Comment by meg-gupta** [comment]
https://github.com/apple/swift/pull/33205 converts ome into a function pass. We support inlining of ossa functions into non-ossa functions in the inliner. The ossa function is converted on demand in the SILCloner. Updated the PR to reflect this, also this can be merged only after https://github.com/apple/swift/pull/33205.

**Comment by meg-gupta** [comment]
@swift-ci test

**Pull request merged**